doc/tutorials/dockerfile.rst

```diff
@@ -175,6 +177,47 @@ For a Dockerfile to work on Binder, it must meet the following requirements:
 
    For more information, and a shell wrapper example, please see the `Dockerfile best practices: ENTRYPOINT  <https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#entrypoint>`_ documentation.
 
+You can build and test your image locally like this.
+
+1. Try building your image.
+
+   .. code-block:: sh
+
+      docker build -t my-image .
+
+2. Try starting a container from the image.
+
+   .. code-block:: sh
+
+      docker run -it --rm -p 8888:8888 my-image jupyter-notebook --ip=0.0.0.0 --port=8888
+
+3. Try starting a container from the image with the additional security requirements.
+
+   .. code-block:: sh
+
+      docker run -it --rm -p 8888:8888 --security-opt=no-new-privileges:true my-image jupyter-notebook --ip=0.0.0.0 --port=8888
```

Comment:
What does this test? Or asked differently: what would have to be in my `Dockerfile` for this check to fail?

Maybe we can add a comment to this step saying "this will fail if you do something like X or Y" to help people. I think `--security-opt=no-new-privileges:true` is something most users of Docker have no idea about (for sure I'd have to google what exactly it does/prevents :D )

Reply:
I'm not sure about this. I ended up thinking this meant one thing, but I then failed to verify that. So instead of letting this stale out I figure I'll remove this point and just leave the rest that isn't complicated.

The idea with this point was to mimic the k8s container's `securityContext` configuration of `allowPrivilegeEscalation: false`, which I think cause a difference between how the docker image is run locally compared to how it is run in k8s via a BinderHub.

But, failing to verify this and being confused about it, let's remove it.